src/generator.cs

```diff
@@ -5899,10 +5923,12 @@ public void Generate (Type type)
 					if (!field_pi.PropertyType.IsValueType) {
 						print ("[CompilerGenerated]");
 						PrintPreserveAttribute (field_pi);
+						PrintAdviceAttribute (field_pi);
```

Comment:
That does not look public, so there's no chance to ever show _this_ advice to the developer. It's also printed later so it will be shown.